### PR TITLE
Upgrade Phpstan to install Symfony's extension

### PR DESCRIPTION
**Pull request by jjanvier**
This extension allows us to identify : 
- undefined Symfony services
- services that are declared private but called publicly from the container

Also, this will be needed as soon as we'll try to reach level 2 of phpstan.

For instance, if have something like:

```php
$someService = $container->get('some-service-id');
$someService->someMethod();
```

Without the extension, phpstan will raise an error because it does not know what is `$someService` and therefore if the method `someMethod()` exists. With the extension, phpstan will detect `$someService` is an instance of the class `SomeAwesomeService` and therefore will be able to check if the method `someMethod` exists.

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,5 @@
+includes:
+    - vendor/phpstan/phpstan-symfony/extension.neon
+parameters:
+    symfony:
+        container_xml_path: %rootDir%/../../../var/cache/dev/appDevDebugProjectContainer.xml
```
Comment (arnolanglade) [inline phpstan.neon]: Should we use the production container?

Comment (jjanvier) [reply phpstan.neon]: no idea 
we'll see what the CI says

Comment (jjanvier) [reply phpstan.neon]: I used the "recommended" configuration https://github.com/phpstan/phpstan-symfony#usage

**Comment by BitOne** [comment]
Does it makes the https://github.com/akeneo/pim-community-dev/blob/master/bin/check-services-instantiability script obsolete?

File: phpstan.neon (hunk 1 above)
Comment (arnolanglade) [reply phpstan.neon]: ok :+1:

**Comment by jjanvier** [comment]
@BitOne hmmm,  I'd say no.
Phpstan checks only the PHP code. It checks the things like `$container->get(...)`.
Your script checks all the service definitions we have declared. And most of them are called through `$container->get(...)`.

BTW, your script currently detects "Found 14 non instantiable services on a total of 2260.". Is it normal? Is it supposed to be launched on the CI?

**Pull request merged**